### ISSUE TITLE
Can't install due to versions conflicts

### ISSUE THREAD

**Issue by sciamannikoo** [ApiGen/ApiGen#844]
If I try to add ApiGet to my project, I get a bunch of problems:
```
Using version ^4.1 for apigen/apigen                                                         
./composer.json has been updated
Loading composer repositories with package information                                                                         Updating dependencies (including require-dev)         Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install apigen/apigen v4.1.2
    - Conclusion: don't install apigen/apigen v4.1.1
    - Conclusion: remove symfony/console v3.2.6
    - Installation request for apigen/apigen ^4.1 -> satisfiable by apigen/apigen[v4.1.0, v4.1.1, v4.1.2].
    - Conclusion: don't install symfony/console v3.2.6
    - apigen/apigen v4.1.0 requires symfony/console ~2.6 -> satisfiable by symfony/console[2.6.x-dev, 2.7.x-dev, 2.8.x-dev, v2.6.0, v2.6.0-BETA1, v2.6.0-BETA2, v2.6.1, v2.6.10, v2.6.11, v2.6.12, v2.6.13, v2.6.2, v2.6.3, v2.6.4, v2.6.5, v2.6.6, v2.6.7, v2.6.8, v2.6.9, v2.7.0, v2.7.0-BETA1, v2.7.0-BETA2, v2.7.1, v2.7.10, v2.7.11, v2.7.12, v2.7.13, v2.7.14, v2.7.15, v2.7.16, v2.7.17, v2.7.18, v2.7.19, v2.7.2, v2.7.20, v2.7.21, v2.7.22, v2.7.23, v2.7.24, v2.7.25, v2.7.26, v2.7.3, v2.7.4, v2.7.5, v2.7.6, v2.7.7, v2.7.8, v2.7.9, v2.8.0, v2.8.0-BETA1, v2.8.1, v2.8.10, v2.8.11, v2.8.12, v2.8.13, v2.8.14, v2.8.15, v2.8.16, v2.8.17, v2.8.18, v2.8.19, v2.8.2, v2.8.3, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v2.8.8, v2.8.9].
    - Can only install one of: symfony/console[2.6.x-dev, v3.2.6].
    - Can only install one of: symfony/console[2.7.x-dev, v3.2.6].
    - Can only install one of: symfony/console[2.8.x-dev, v3.2.6].
(...a lot more lines: hiding...)
    - Can only install one of: symfony/console[v2.8.8, v3.2.6].
    - Can only install one of: symfony/console[v2.8.9, v3.2.6].
    - Installation request for symfony/console (locked at v3.2.6) -> satisfiable by symfony/console[v3.2.6].
```

Currently, this why `symfony/console` is used:
```
otgs/changelog      0.1.0  requires  symfony/console (^3.2)         
scrutinizer/ocular  1.3.1  requires  symfony/console (^2.0.5|~3.0)  
sebastian/phpcpd    3.0.0  requires  symfony/console (^3.0)  
```

While I could fix `otgs/changelog`, as I own this project, by using the same approach used by `scrutinizer/ocular`, I can't tell Sebastian Bergmann to change `sebastian/phpcpd` as he may need features which are only available in version 3.0 of `symfony/console`.

Any idea how I can get around this issue?

I need to use ApiGen in a build process, so I've tried these steps:

- delete `composer.lock` file (otherwise composer complain because `Installation request for symfony/console (locked at v3.2.6) -> satisfiable by symfony/console[v3.2.6]`)
- run `composer install --no-dev`
- install ApiGen as a non-dev library (as the above are dev libraries and the conflicting dependencies would be installed again).

This solution still don't work, as, despite `symfony/console` not being in the "vendor" directory anymore, I still get the initial error.

Moreover, this solution, if it would work, would considerably slow down the CI job and, to be honest, looks quite a dirty solution.

There is any clean way I can get around this issue?

**Comment by TomasVotruba** [comment]
The main focus is on 5.0 branch for now, no support for 4.x is planned.

This might help: https://github.com/ApiGen/ApiGen/issues/843#issuecomment-294853422

**Comment by sciamannikoo** [comment]
I was asking for a solution to my problem, to for a fix in the library.

Never mind.

I guess I can live without ApiGen and either look to some other tool or wait for 5.0, provided it will fix the issue I'm facing.

Thanks anyway.

**Comment by TomasVotruba** [comment]
Apigen 5 fixes your issue, since it requires Symfony 3.2 components.

At the moment, there is old reflection package, **that works properly only on PHP 5.4 code**. Higher version can break it. That's why the main goal is to replace it: https://github.com/ApiGen/ApiGen/issues/817, before any other change.

For more see https://github.com/ApiGen/ApiGen/issues/782 and  https://github.com/ApiGen/ApiGen/issues/779
I explain reasons there in more detailed way.

**Comment by sciamannikoo** [comment]
Thanks for the clarification.

I'll keep an eye on the progresses.
